akka-actor-typed-tests/src/test/scala/akka/actor/typed/ActorRefIgnoreSpec.scala

```diff
@@ -0,0 +1,150 @@
+
+import java.util.concurrent.TimeoutException
+
+import akka.actor.typed.scaladsl.Behaviors
+import org.scalatest.wordspec.AnyWordSpecLike
+import org.scalatest.concurrent.PatienceConfiguration.{ Timeout => PatienceTimeout }
+import akka.actor.testkit.typed.scaladsl.{ ScalaTestWithActorTestKit, TestProbe }
+import akka.actor.typed.scaladsl.adapter._
+import akka.actor.typed.scaladsl.AskPattern._
+import akka.util.Timeout
+
+import scala.concurrent.duration._
+import scala.util.{ Failure, Success }
+
+class ActorRefIgnoreSpec extends ScalaTestWithActorTestKit() with AnyWordSpecLike {
+
+  case class Request(replyTo: ActorRef[Int])
+
+  // this Actor behavior receives simple request and answers back total number of
+  // messages it received so far
+  val askMeActorBehavior: Behavior[Request] = {
+    def internalBehavior(counter: Int): Behavior[Request] =
+      Behaviors.receiveMessage[Request] {
+        case Request(replyTo) =>
+          val newCounter = counter + 1
+          replyTo ! newCounter
+          internalBehavior(newCounter)
+      }
+
+    internalBehavior(0)
+  }
+
+  /**
+   * This actor sends a ask to 'askMeRef' at bootstrap and forward the answer to the probe.
+   * We will use it through out this test.
+   */
+  def behavior(askMeRef: ActorRef[Request], probe: TestProbe[Int]) = Behaviors.setup[Int] { context =>
+    implicit val timeout: Timeout = 1.second
+
+    // send a message to interactWithRef
+    context.ask(askMeRef, Request) {
+      case Success(res) => res
+      case Failure(ex)  => throw ex
+    }
+
+    Behaviors.receiveMessage { num =>
+      // receive response from interactWithRef and sent to prob
+      probe.ref ! num
+      Behaviors.same
+    }
+  }
+
+  "IgnoreActorRef instance" should {
+
+    "ignore all incoming messages" in {
+
+      val askMeRef = testKit.spawn(askMeActorBehavior)
+
+      val probe = testKit.createTestProbe[Int]("response-probe")
+      askMeRef ! Request(probe.ref)
+      probe.expectMessage(1)
+
+      // this is more a compile-time proof
+      // since the reply is ignored, we can't check that a message was sent to it
+      askMeRef ! Request(testKit.system.ignoreRef)
+      probe.expectNoMessage()
+
+      // but we do check that the counter has increased when we used the ActorRef.ignore
+      askMeRef ! Request(probe.ref)
+      probe.expectMessage(3)
+    }
+
+    // this is kind of obvious, the Future won't complete because the ignoreRef is used
+    "make a Future timeout when used in a 'ask'" in {
+
+      implicit val timeout: Timeout = 500.millis
+      val askMeRef = testKit.spawn(askMeActorBehavior)
+
+      val failedAsk =
+        askMeRef
+          .ask { _: ActorRef[Request] =>
+            Request(testKit.system.ignoreRef) // <- pass the ignoreRef instead, so Future never completes
+          }
+          .failed
+          .futureValue(PatienceTimeout(1.second))
+
+      eventually {
+        failedAsk shouldBe a[TimeoutException]
+      }
```

Comment:
`eventually` means that it will retry the block, but here there is nothing to retry. The Future has already been completed, and waited for with `futureValue`. I think this works fine without eventually.

Reply:
correct, I will remove it